lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -164,7 +164,7 @@ protected function getType(&$value)
                 return self::T_STRING;
 
             // Recognize identifiers
-            case (ctype_alpha($value[0]) || $value[0] === '_'):
+            case (ctype_alpha($value[0]) || $value[0] === '_' || $value[0] === '\\'):
```

Comment:
Adding a `\Doctrine\Tests\ORM\Query\LexerTest::testScannerRecognizesIdentifier` test that also covers leading `\` in FQCNs; plus a new \Doctrine\Tests\ORM\Query\ParserTest that asserts `AbstractSchemaName` from BCNF will have leading `\` stripped.